### PR TITLE
[W4.5c][W10-A1]Joshua Lee Kai Sheng

### PR DESCRIPTION
**Pull request by jlks96**
Completed LO-Exceptions

### CODE REVIEW

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -84,10 +85,13 @@ private void runCommandLoopUntilExitCommand() {
...
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
-
+            try {
+                CommandResult result = executeCommand(command);
+                recordResult(result);
+                ui.showResultToUser(result);
+            } catch (StorageOperationException e) {
+                start(new String[0]);//restart the program
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java:93]: This is one of the ways to handle the exception. Think what would make the user experience better in terms of not losing the data entered.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -84,10 +85,13 @@ private void runCommandLoopUntilExitCommand() {
...
             command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
-
+            try {
+                CommandResult result = executeCommand(command);
+                recordResult(result);
+                ui.showResultToUser(result);
+            } catch (StorageOperationException e) {
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java:92]: In general, it is good to handle the exception closer to where it occurs.

File: src/seedu/addressbook/common/Messages.java [hunk 3]
```diff
@@ -15,4 +15,5 @@
             "java seedu.addressbook.Main [STORAGE_FILE_PATH]";
     public static final String MESSAGE_WELCOME = "Welcome to your Address Book!";
     public static final String MESSAGE_USING_STORAGE_FILE = "Using storage file : %1$s";
+    public static final String MESSAGE_UNCHECK_READ_ONLY = "Please uncheck the \" Read Only \" option for the storage file!";
```
Comment (okkhoy) [inline src/seedu/addressbook/common/Messages.java:18]: :+1:  for separating out the message to where it belongs!

**Comment by okkhoy** [comment]
@jlks96 
Some comments added. Please ack & close the PR after reading comments.

**Comment by jlks96** [comment]
Acknowledged. Thanks

**Pull request closed without merging**